lib/metrics/baseCollector.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 Huawei Cloud Computing Technologies Co., Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ModuleIndex stores the indicators of each module
+type ModuleIndex struct {
+	MetricsMap  map[string]interface{}
+	LabelValues map[string]string
+}
+
+type Metric struct {
+	HelpMap map[string]string
+	Labels  []string
+}
+
+func NewDesc(subsystem, name, help string, labels []string) *prometheus.Desc {
+	return prometheus.NewDesc(
+		prometheus.BuildFQName("", subsystem, name),
+		help,
+		labels,
+		nil,
+	)
+}
+
+type BaseCollector struct {
+	AllModulesDesc map[string]map[string]*prometheus.Desc
+	IndexRegistry  map[string]*Metric
+}
+
+func NewBaseCollector(comIndexRegistry map[string]*Metric) *BaseCollector {
+	c := &BaseCollector{
+		AllModulesDesc: make(map[string]map[string]*prometheus.Desc),
+		IndexRegistry:  comIndexRegistry,
+	}
+
+	return c
+}
+
+func (c *BaseCollector) Collect(ch chan<- prometheus.Metric) {
```

Comment:
How does the `collector` react when the `/metrics` API is called? no specific logic here!

Reply:
This piece is an empty implementation, the specific logic is in handler_metrics.go